.github/workflows/ci.yml

```diff
@@ -3,75 +3,50 @@ name: CI/CD
 on:
   push:
     branches:
-      - master
-      - master-py27
+      - 'master*'
   pull_request:
   # Run daily at 0:01 UTC
   schedule:
   - cron:  '1 0 * * *'
 
 jobs:
   linter:
-    runs-on: ${{ matrix.os }}
-    strategy:
-      matrix:
-        os: [ubuntu-latest]
-        python-version: [3.7]
-    name: linter - flake8 - python ${{ matrix.python-version }}
+    runs-on: ubuntu-latest
+    name: pre-commit (linter)
```

Comment:
This runs black, flake8, and a few more checks,  provides nice error messages, and has custom caching built in, so it's lightning fast.